core/Columns/Dimension.php

```diff
@@ -853,4 +882,36 @@ public function getVersion()
         return $this->columnType;
     }
 
+
+    /**
+     * Returns an idaction value to match an idaction column by searching log_action, if $matchType is
+     * SegmentExpression::MATCH_EQUAL or SegmentExpression::MATCH_NOT_EQUAL. This method is used
+     * to optimize segment conditions involving idaction queries, avoiding a join by querying the log_action
+     * table beforehand.
+     *
+     * Should be used as the $sqlFilter property for idaction dimensions that use `ActionNameJoin`.
+     *
+     * @param string $value the value in the segment condition
+     * @param string $sqlField the table column of the segment condition
+     * @param string $matchType the SegmentExpression match type, eg, `SegmentExpression::MATCH_NOT_EQUAL`
+     * @param string $segmentName the name of the segment, ie, `pageUrl`
+     * @return array|null|string
+     */
+    public function getOptimizedIdActionSqlMatch($value, $sqlField, $matchType, $segmentName, $isUrlProtocolAbsentInDb = false)
```

Comment:
It originally did something different. And it's specifically for dimensions to customize how the dimension's segment is converted to SQL. But it can be moved, I don't have a strong opinion on this.

Reply:
@sgiehl do you still believe this should be moved?

Reply:
Actually, yes. It only uses methods from `TableLogAction`, so logically it feels more that this would be a better place for the method. Or do you think there could be the requirement that a specific dimension would need to overwrite this method?